### PR TITLE
Fix warning-related issues

### PR DESCRIPTION
**Pull request by williamwen42**
Address https://github.com/pytorch/torchdynamo/issues/948 and fix related issues:
- replace the use of the `warnings` module in the main code
- replace root logging with module-scoped logging
- use function `logging.warning` instead of deprecated `logging.warn`.

### CODE REVIEW

**Review by anijain2305 — APPROVED** [review]

File: benchmarks/common.py [hunk 1]
```diff
@@ -1855,7 +1855,7 @@ def main(runner, original_dir=None):
                     batch_size=batch_size,
                 )
             except NotImplementedError:
-                logging.warn(f"{args.only} failed to load")
+                logging.warning(f"{args.only} failed to load")
```
Comment (jansel) [inline benchmarks/common.py:1858]: Introduce a `log = loggging.getLogger(__name__)` at top of file.

File: benchmarks/huggingface.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def get_sequence_length(model_cls, model_name):
     ) or model_name in ("DistillGPT2", "GoogleFnet", "YituTechConvBert", "CamemBert"):
         seq_length = 512
     else:
-        logging.warn(
+        logging.warning(
```
Comment (jansel) [inline benchmarks/huggingface.py:163]: Introduce a `log = loggging.getLogger(__name__)` at top of file.

File: benchmarks/microbenchmarks/operator_inp_utils.py [hunk 3]
```diff
@@ -256,7 +256,7 @@ def get_inputs_for_operator(
         ), f"Could not find {operator}, must provide overload"
 
         if "embedding" in str(operator):
-            logging.warn("Embedding inputs NYI, input data cannot be randomized")
+            logging.warning("Embedding inputs NYI, input data cannot be randomized")
```
Comment (jansel) [inline benchmarks/microbenchmarks/operator_inp_utils.py:259]: Introduce a `log = loggging.getLogger(__name__)` at top of file.

**Review by jansel — CHANGES_REQUESTED** [review]
What was the underlying issue?

**Comment by williamwen42** [comment]
True underlying issue is unclear since @ezyang wasn't able to give me an exact repro for the issue. I have sample testing scripts that show that the problem could potentially happen, but for now, we're just making preemptive fixes.

File: benchmarks/huggingface.py [hunk 4]
```diff
@@ -11,6 +11,8 @@
 from common import BenchmarkRunner
 from common import main
 
+log = logging.getLogger(__name__)
```
Comment (jansel) [inline benchmarks/huggingface.py:14]: This should go after the imports.

**Review by jansel — APPROVED** [review]

**Pull request merged**